### PR TITLE
[api] reply to comment function

### PR DESCRIPTION
**Pull request by lethliel**
* implemented function to reply to parent comments
* changed tests to satisfy new validator
* added test

### CODE REVIEW

File: src/api/spec/models/comment_spec.rb [hunk 1]
```diff
@@ -2,9 +2,9 @@
 
 RSpec.describe Comment do
   let(:comment_package) { create(:comment_package) }
-  let(:comment_package_with_parent) { create(:comment_package, parent: comment_package) }
-  let(:comment_package_with_parent_2) { create(:comment_package, parent: comment_package) }
-  let(:comment_package_with_grandparent) { create(:comment_package, parent: comment_package_with_parent) }
+  let(:comment_package_with_parent) { create(:comment_package, parent: comment_package, commentable: comment_package.commentable) }
```
Comment (hennevogel) [inline src/api/spec/models/comment_spec.rb]: Why do you need to do this?

Comment (lethliel) [inline src/api/spec/models/comment_spec.rb]: It is needed for the comment model to validate the parent_id properly during the test.

Comment (hennevogel) [inline src/api/spec/models/comment_spec.rb]: But this is what the factory already does

```
commentable { create(:package) }
```

Comment (hennevogel) [inline src/api/spec/models/comment_spec.rb]: Ah we need a new factory. Let's do this afterward

**Pull request merged**